### ISSUE TITLE
Question : Problem creating custom GUI in C#

### ISSUE THREAD

**Issue by Horopter** [Emotiv/cortex-example#56]
Operating system: Windows 8.1
Last upstream commit (run `git log --author="Robert Lord" | head -n 1`): Latest Commit
language version:(nodejs version, ruby version ....) C#, VS2017
Cortex version:  Latest Download from Emotiv Website

---
I have created a win form in the following format.
The data recording stops after 7 seconds. 
Most of the time, there is a failure in setting up of a session.
Can you please let me know why? 

using System;
using CortexAccess;
using System.Threading;
using System.Collections;
using System.Windows.Forms;

namespace EEGLogViewer
{
    public partial class EEGLogView : Form
    {
        public EEGLogView()
        {
            CheckForIllegalCrossThreadCalls = false;
            InitializeComponent();
        }

        private void EEGLogView_Load(object sender, EventArgs e)
        {
            const string Username = "<filled>";
            const string Password = "<filled>";
            const string LicenseId = "<filled>";
            const int DebitNumber = 2; // default number of debit

            Process p = new Process();

            // Register Event
            p.OnEEGDataReceived += OnEEGDataReceived;
            p.SessionCtr.OnSubcribeEEGOK += OnEEGDataReceived;

            if (string.IsNullOrEmpty(p.GetUserLogin()))
            {
                p.Login(Username, Password);
                Thread.Sleep(5000);
            }
            // Show username login
            Notice.Text = "Username :" + p.GetUserLogin();
            
            if (p.AccessCtr.IsLogin)
            {
                p.Authorize(LicenseId, DebitNumber);
                Thread.Sleep(5000);
            }
            if (!p.IsHeadsetConnected())
            {
                p.QueryHeadset();
                Thread.Sleep(5000);
            }
            if (!p.IsCreateSession)
            {
                p.CreateSession();
                Thread.Sleep(5000);
            }
            if (p.IsCreateSession)
            {
                p.SubcribeData("eeg");
                Thread.Sleep(5000);
            }
            else
            {
                Notice.Text = "Remote Host refused to create a session. Good bye!";
            }

            // Unsubcribe stream
            p.UnSubcribeData("eeg");
            Notice.Text += "  EEG unsubscribed!";
            Thread.Sleep(5000);
        }

        private void OnEEGDataReceived(object sender, ArrayList eegData)
        {
            Notice.Text = DateTime.Now.ToString("yyyy, MM, dd, hh, mm, ss");
            textBox1.Text = eegData[0].ToString();
            textBox2.Text = eegData[1].ToString();
            textBox3.Text = eegData[2].ToString();
            textBox4.Text = eegData[3].ToString();
            textBox5.Text = eegData[4].ToString();
            textBox6.Text = eegData[5].ToString();
            textBox7.Text = eegData[6].ToString();
            textBox8.Text = eegData[7].ToString();
            textBox9.Text = eegData[8].ToString();
            textBox10.Text = eegData[9].ToString();
            textBox11.Text = eegData[10].ToString();
            textBox12.Text = eegData[11].ToString();
            textBox13.Text = eegData[12].ToString();
            textBox14.Text = eegData[13].ToString();
            textBox15.Text = eegData[14].ToString();
            textBox16.Text = eegData[15].ToString();
            textBox17.Text = eegData[16].ToString();
        }
    }
}

**Comment by patdotme** [comment]
hi @Horopter what's your EmotivID and do you have a PRO license to access the EEG data?
https://www.emotiv.com/developer/

**Comment by Horopter** [comment]
Yes bro, I do have a PRO license. I don't think I should reveal my license number or ID, nothing personal, I am not allowed to. Rest assured, it's a software problem. Here's a 5 second video clip after which the numbers stagnate. 

It'd be great if this problem gets resolved by tomorrow.

[20181111_045359.zip](https://github.com/Emotiv/cortex-example/files/2569970/20181111_045359.zip)

**Comment by patdotme** [comment]
Without the full source code of your project I could only guess it is related to this:
```
        if (p.IsCreateSession)
        {
            p.SubcribeData("eeg");
            Thread.Sleep(5000);
        }
        else
        {
            Notice.Text = "Remote Host refused to create a session. Good bye!";
        }

        // Unsubcribe stream
        p.UnSubcribeData("eeg");
        Notice.Text += "  EEG unsubscribed!";
        Thread.Sleep(5000);
```
After subscribed to the EEG data stream successfully the thread sleeps for 5 seconds and when it wakes up again, it unsubscribes the EEG data stream and hence the incoming data stops.

Also both EEGLogView_Load and OnEEGDataReceived are writing to the same Notice.Text but I don't see the timestamp being displayed in the video.

And sorry that we cannot guarantee a 24-hour respond time on the Github issues at the moment.

**Comment by Horopter** [comment]
[www.github.com/Horopter/cortex-example](www.github.com/Horopter/cortex-example) is where you can find my fork of the repo. Please feel free to check out the full source.
:)
Thank you.
